lte/gateway/c/core/oai/test/sgw_s8_task/sgw_csr.cpp

```diff
@@ -0,0 +1,354 @@
+    bdestroy_wrapper(&config->config_file);
+    free(config);
+  }
+};
+
+// TC validates creation of UE context on reception of Create Session Req
+TEST_F(SgwS8Config, create_ue_context_cs_req) {
+  sgw_state_init(false, config);
+  sgw_state_t* sgw_state = get_sgw_state(false);
+
+  spgw_ue_context_t* ue_context_p = NULL;
+  mme_sgw_tunnel_t sgw_s11_tunnel = {0};
+  sgw_s11_tunnel.local_teid       = sgw_s8_generate_new_cp_teid();
+  EXPECT_EQ(
+      sgw_update_teid_in_ue_context(
+          sgw_state, imsi64, sgw_s11_tunnel.local_teid),
+      RETURNok);
+  EXPECT_EQ(
+      hashtable_ts_get(
+          sgw_state->imsi_ue_context_htbl, (const hash_key_t) imsi64,
+          reinterpret_cast<void**>(&ue_context_p)),
+      HASH_TABLE_OK);
+
+  sgw_state_exit();
+}
+
+// TC validates creation of pdn session on reception of Create Session Req
+TEST_F(SgwS8Config, create_pdn_session_cs_req) {
+  sgw_state_init(false, config);
+  sgw_state_t* sgw_state = get_sgw_state(false);
+
+  spgw_ue_context_t* ue_context_p = NULL;
+  mme_sgw_tunnel_t sgw_s11_tunnel = {0};
+  sgw_s11_tunnel.local_teid       = sgw_s8_generate_new_cp_teid();
+
+  EXPECT_EQ(
+      sgw_update_teid_in_ue_context(
+          sgw_state, imsi64, sgw_s11_tunnel.local_teid),
+      RETURNok);
+  EXPECT_EQ(
+      hashtable_ts_get(
+          sgw_state->imsi_ue_context_htbl, (const hash_key_t) imsi64,
+          reinterpret_cast<void**>(&ue_context_p)),
+      HASH_TABLE_OK);
+
+  sgw_eps_bearer_context_information_t* sgw_pdn_session = NULL;
+  sgw_pdn_session = sgw_create_bearer_context_information_in_collection(
+      sgw_s11_tunnel.local_teid);
+  EXPECT_TRUE(sgw_pdn_session != nullptr);
+
+  sgw_state_exit();
+}
+
+// TC validates creation of bearer context on reception of Create Session Req
+TEST_F(SgwS8Config, create_bearer_within_pdn_session_cs_req) {
+  sgw_state_init(false, config);
+  sgw_state_t* sgw_state = get_sgw_state(false);
+
+  spgw_ue_context_t* ue_context_p = NULL;
+  mme_sgw_tunnel_t sgw_s11_tunnel = {0};
+  sgw_s11_tunnel.local_teid       = sgw_s8_generate_new_cp_teid();
+  EXPECT_EQ(
+      sgw_update_teid_in_ue_context(
+          sgw_state, imsi64, sgw_s11_tunnel.local_teid),
+      RETURNok);
+  EXPECT_EQ(
+      hashtable_ts_get(
+          sgw_state->imsi_ue_context_htbl, (const hash_key_t) imsi64,
+          reinterpret_cast<void**>(&ue_context_p)),
+      HASH_TABLE_OK);
+
+  sgw_eps_bearer_context_information_t* sgw_pdn_session = NULL;
+  sgw_pdn_session = sgw_create_bearer_context_information_in_collection(
+      sgw_s11_tunnel.local_teid);
+  EXPECT_TRUE(sgw_pdn_session != nullptr);
+  itti_s11_create_session_request_t session_req = {0};
+  fill_itti_csreq(&session_req);
+  sgw_s11_tunnel.remote_teid = session_req.sender_fteid_for_cp.teid;
+  EXPECT_EQ(
+      sgw_update_bearer_context_information_on_csreq(
+          sgw_state, sgw_pdn_session, sgw_s11_tunnel, &session_req, imsi64),
+      RETURNok);
+  sgw_state_exit();
+}
```

Comment:
I think in order to reduce the amount of code/test and put more effort on more complex test, lets keep only create_bearer_within_pdn_session_cs_req which includes, create_pdn_session_cs_req and create_ue_context_cs_req

We can set it as `create session on cs rsp`